src/main/java/com/dsrm/dsrmbackend/services/impl/UserServiceImpl.java

```diff
@@ -35,4 +36,9 @@ public Page<User> getUsers(Pageable pageable){
         return userRepo.findAll(pageable);
     }
 
+    @Override
+    public Page<User> getUsersWithRoles(Role roles, Pageable pageable) {
+        return userRepo.findUserByRoles(roles,pageable);
```

Comment:
we want to use specification instead of jpa methods because we are doing filtering. The method should look like this
```
    public Page<User> getUsers(Pageable pageable, boolean isPending){
        Specification<User> userSpecification = Specification.where(null);
        if (isPending) {
            userSpecification = userSpecification.and(
                    (Specification<User>) (root, query, criteriaBuilder) -> criteriaBuilder.isEmpty(root.get("roles"))
            );
        }
        return userRepo.findAll(userSpecification, pageable);
    }
```
Reference from task https://spring.io/blog/2011/04/26/advanced-spring-data-jpa-specifications-and-querydsl/